src/PSRule.Rules.Azure/rules/Azure.VNET.Rule.ps1

```diff
@@ -107,4 +107,37 @@ Rule 'Azure.VNET.SubnetName' -Ref 'AZR-000267' -Type 'Microsoft.Network/virtualN
     }
 }
 
+# Synopsis: VNETs with a GatewaySubnet should have an AzureBastionSubnet to allow for out of band remote access to VMs.
+Rule 'Azure.VNET.BastionSubnet' -Ref 'AZR-000314' -Type 'Microsoft.Network/virtualNetworks' -If { IsGatewaySubnet } -Tag @{ release = 'GA'; ruleSet = '2022_12' } {
+    $subnets = @(GetVirtualNetworkSubnets)
+    if ($subnets | Where-Object { $_.name -eq 'AzureBastionSubnet' }) { $Assert.Pass() } else { $Assert.Fail().Reason($LocalizedData.BastionSubnetNotFound, $TargetObject.name) }
```

Comment:
If `$subnets` is just an array of strings you can simplify this like:

```suggestion
    $Assert.In($subnets, '.', @('AzureBastionSubnet')).Reason($LocalizedData.BastionSubnetNotFound, $PSRule.TargetName) }
```

Also you may not need to include the name of the VNET in the reason, because it is already included in PSRule output.

Reply:
Better, also preferably to use assertion methods and variables provided by PSRule.